### PR TITLE
support other csv delimiters for text

### PR DESCRIPTION
**Pull request by jtbr**
Adds support for non-comma delimiters in csv files, as well as for naming columns in csv files without headers. This is especially important for text, since commas are often found in text.
Documentation is also updated.

### CODE REVIEW

File: fastai/data_block.py [hunk 1]
```diff
@@ -121,10 +121,11 @@ def from_df(cls, df:DataFrame, path:PathOrStr='.', cols:IntsOrStrs=0, processor:
         return res
 
     @classmethod
-    def from_csv(cls, path:PathOrStr, csv_name:str, cols:IntsOrStrs=0, header:str='infer', 
-                 processor:PreProcessors=None, **kwargs)->'ItemList':
-        "Create an `ItemList` in `path` from the inputs in the `cols` of `path/csv_name` opened with `header`."
-        df = pd.read_csv(Path(path)/csv_name, header=header)
+    def from_csv(cls, path:PathOrStr, csv_name:str, cols:IntsOrStrs=0, delimiter:str=None, header:str='infer', 
+                 col_names:Optional[Collection[str]]=None, processor:PreProcessors=None, **kwargs)->'ItemList':
```
Comment (sgugger) [inline fastai/data_block.py]: The `cols` argument can already either be a collection of strings that are the column names, so I don't want to add another one that would be redundant. You can check if `cols` is a list of strings with `is_listy(cols) and isinstance(cols[0],str)` for instance.

File: fastai/data_block.py [hunk 2]
```diff
@@ -121,10 +121,11 @@ def from_df(cls, df:DataFrame, path:PathOrStr='.', cols:IntsOrStrs=0, processor:
...
+        """Create an `ItemList` in `path` from the inputs in the `cols` of `path/csv_name` opened with `header`.
```
Comment (sgugger) [inline fastai/data_block.py]: Let's leave it a the first line (fastai style guide) and the second line of this doc string should go in the doc notebook).

File: fastai/text/data.py [hunk 3]
```diff
@@ -204,15 +204,15 @@ def from_df(cls, path:PathOrStr, train_df:DataFrame, valid_df:DataFrame, test_df
 
     @classmethod
     def from_csv(cls, path:PathOrStr, csv_name, valid_pct:float=0.2, test:Optional[str]=None,
-                 tokenizer:Tokenizer=None, vocab:Vocab=None, classes:Collection[str]=None, header = 'infer', text_cols:IntsOrStrs=1,
-                 label_cols:IntsOrStrs=0, label_delim:str=None, chunksize:int=10000, max_vocab:int=60000,
-                min_freq:int=2, mark_fields:bool=False, **kwargs) -> DataBunch:
+                 tokenizer:Tokenizer=None, vocab:Vocab=None, classes:Collection[str]=None, delimiter:str=None, header='infer',
+                 col_names:Optional[Collection[str]]=None, text_cols:IntsOrStrs=1, label_cols:IntsOrStrs=0, label_delim:str=None,
```
Comment (sgugger) [inline fastai/text/data.py]: No `col_names` here either
You should also change `ImageDataBunch.from_csv` to add the `delimiter` argument.

**Comment by sgugger** [comment]
Thanks. There is one problem with this PR outlined in the comments I left and a few things to tweak, but it seems like a useful functionality.

File: fastai/data_block.py (hunk 1 above)
Comment (jtbr) [reply fastai/data_block.py]: The case where col_names is useful is where the csv has no header, so pandas doesn't know the column names. I don't think cols can overcome this (of course you can work around it by using only indices, which might be enough if there aren't many columns). Your call. 

I'll work on the other changes you mention.

Comment (sgugger) [reply fastai/data_block.py]: Oh right, cols would only contain the names of the inputs columns, not the outputs. If there are no col names, though, you have to know in advance what indexes your inputs are, so let's use integers in cols in this case.
I just don't want to have yet another argument if it's not necessary (`delimiter` is useful however).

**Comment by jtbr** [comment]
I've made the changes you requested.

**Comment by sgugger** [comment]
Looking good, thanks a lot!

**Pull request merged**